src/ast/mod.rs

```diff
@@ -124,6 +124,12 @@ impl<'a> Display for Primitive {
     }
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct Branch<'a> {
```

Comment:
I know we don't have a lot of docs but could you add an small description of this structure?